ibis/backends/base/__init__.py

```diff
@@ -153,6 +153,20 @@ def list_tables(self, like: str = None, database: str = None) -> list[str]:
             The list of the table names that match the pattern `like`.
         """
 
+    def exists_table(self, name: str, database: str = None) -> bool:
+        """
+        Return whether a table name exists in the database.
+
+        Deprecated in Ibis 2.0. Use `name in client.list_tables()` instead.
+        """
+        warnings.warn(
+            '`client.exists_table(name)` is deprecated, and will be '
+            'removed in a future version of Ibis. Use '
+            '`name in client.list_tables()` instead.',
+            FutureWarning,
+        )
+        return name in self.client.list_tables(database=database)
```

Comment:
Should this be implemented the way it was before as `len(self.client.list_tables(like=name, database=database)) > 0`?

Reply:
`name` won't match tables in other databases, so I don't think you can use `in` here. Since this isn't doing something trivial, should we reconsider deprecating it?

Reply:
I don't fully understand what's Impala doing. I don't think other backends are affected.

We've got a syntax for checking other databases, `name in con.list_tables(database='foo')`. And I think it's better than this magic failing the test. Just the explicit syntax doesn't seem to work in Impala (or there was an error in my code when I tried that, but I don't think so). I'm using the previous syntax of checking the list of tables with the `like` param for now, so we can get tests passing.

Reply:
> I don't fully understand what's Impala doing. I don't think other backends are affected.

Did you read [my comment](https://github.com/ibis-project/ibis/pull/2905#issuecomment-925191316)?

Reply:
> And I think it's better than this magic failing the test.

Are you referring to the regular expression splitting?

> Just the explicit syntax doesn't seem to work in Impala

Are you sure about that? Here's an example that seems to work just fine:

```
(Pdb) list
265
266         table = con.parquet_file(hdfs_path)
267
268         name = table.op().name
269         breakpoint()
270  ->     assert con.exists_table(name) is True
271         con.close()
272
273         assert not con.exists_table(name)
274
275
(Pdb) con.list_tables('__ibis_tmp_81c73096e0924eafba359d2a1de85f85', database='__ibis_tmp')
['__ibis_tmp_81c73096e0924eafba359d2a1de85f85']
```

Reply:
Mhh, I see, you're right. I didn't fully understand the problem, now it's clear. Then, what's different in Impala is that other backends don't support fully qualified table names, as Impala handles in `list_tables`. Not sure why the test wasn't working when I splittted the table myself in the test, I guess I missed something.

In any case, I think this PR is fine now. Or is there anything else you would change?